cdap-ui/app/cdap/components/Popover/index.js

```diff
@@ -140,7 +142,12 @@ export default class Popover extends Component {
           })}
           onClick={this.handleBubbleEvent}
         >
-          {this.props.children}
+          {
+            this.props.injectOnToggle ?
```

Comment:
I'm not sure why we need this?

Reply:
Today when we use `Popover` and add custom components as its children it renders them anyway (with display none) whether the popover is shown or not. This will prevent the component to be rendered until the popover is shown.